src/PackageProvider/LazyPackageProviderDetectionFactory.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Laminas\ComponentInstaller\PackageProvider;
+
+use Composer\Composer;
+use Composer\Installer\PackageEvent;
+
+final class LazyPackageProviderDetectionFactory implements PackageProviderDetectionFactoryInterface
```

Comment:
Why is laziness a factor/necessary? Can we document that? :thinking:

Reply:
We had a lazy callable beforehand. I prefer having a class over a lazy callable and thus created this.

Afaik composer plugins are always activated but we do not need the package provider until a package is being installed/updated/removed on which we have to interact.

So mainly minor performance reasons along with that composer v1/v2 stuff. But I do remove composer v1 in another PR and thus maybe we can even remove this.

Reply:
I dropped it now.